pkg/statistics/handle/usage/predicate_column.go

```diff
@@ -159,6 +166,33 @@ func GetPredicateColumns(sctx sessionctx.Context, tableID int64) ([]int64, error
 	return columnIDs, nil
 }
 
+// cleanupDroppedColumnStatsUsage deletes the column stats usage information whose column is dropped.
+func cleanupDroppedColumnStatsUsage(sctx sessionctx.Context, tableID int64) error {
+	is := sctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+	table, ok := is.TableByID(tableID)
+	if !ok {
+		// Usually, it should not happen.
+		// But if it happens, we can safely do nothing.
+		return nil
+	}
+	allColumns := table.Meta().Columns
+	// Due to SQL limitations, column IDs must be converted to strings for proper escaping in the query :(
+	columnIDs := make([]string, 0, len(allColumns))
+	for _, col := range allColumns {
+		columnIDs = append(columnIDs, fmt.Sprintf("%d", col.ID))
+	}
+
+	// Delete the column stats usage information whose column is dropped.
+	_, _, err := utilstats.ExecRows(
```

Comment:
Do we need to add ```ActionDropColumn```  event to delete the column stats usage information ?

Reply:
make sense

Reply:
![img_v3_02bd_3a5aecb0-4bb4-49e7-927d-8edd1438103g](https://github.com/pingcap/tidb/assets/29879298/713ef9c6-88df-4989-ab7e-afa554fd2a74)
There is a reason why we didn't use the DDL event.